packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -451,11 +452,16 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
               )
 
               if (needsInterop === undefined) {
-                config.logger.error(
-                  colors.red(
-                    `Vite Error, ${url} optimized info should be defined`
+                // Non-entry dynamic imports from dependencies will reach here as there isn't
+                // optimize info for them, but they don't need es interop. If the request isn't
+                // a dynamic import, then it is an internal Vite error
+                if (!file.match(optimizedDepDynamicRE)) {
```

Comment:
Is this line placed here instead of at

https://github.com/vitejs/vite/blob/b4a0b002549cec2345042c0efd39527533500265/packages/vite/src/node/plugins/importAnalysis.ts#L440

because there is a possibility that a `FooBar-abcd1234.js` entry exist in the optimize metadata?

Reply:
Yes, that is what I'm afraid of. And why I added this comment in the description:
> I think later we could save in the metadata the names of all non-entry outputs, chunks, and dynamic imports, so we can do these checks against the real file names.